src/main/java/org/folio/circulation/services/DeclaredLostFeeFineService.java

```diff
@@ -0,0 +1,146 @@
+      .thenApply(result -> refuseWhenFeeFineOfTypeNotFound(result, LOST_ITEM_FEE))
+      .thenApply(result -> refuseWhenFeeFineOfTypeNotFound(result, LOST_ITEM_PROCESSING_FEE))
+      .thenApply(feeFinesResult -> feeFinesResult.next(feeFines -> {
+          final LostItemPolicy policy = loan.getLostItemPolicy();
+          final List<AccountStorageRepresentation> accountsToCreate = new ArrayList<>();
+
+          final FeeFine lostItemFee = getFeeFineOfType(feeFines, LOST_ITEM_FEE);
+          final FeeFine lostItemProcessingFee = getFeeFineOfType(feeFines, LOST_ITEM_PROCESSING_FEE);
+
+          if (shouldChargeItemFee(policy)) {
+            log.debug("Charging lost item fee");
+
+            accountsToCreate.add(new AccountStorageRepresentation(loan, loan.getItem(),
+              owner, lostItemFee, policy.getChargeAmountItem().getAmount()));
+          }
+
+          if (shouldChargeProcessingFee(policy)) {
+            log.debug("Charging lost item processing fee");
+
+            accountsToCreate.add(new AccountStorageRepresentation(loan, loan.getItem(),
+              owner, lostItemProcessingFee, policy.getLostItemProcessingFee()));
+          }
+
+          log.debug("Total accounts created {}", accountsToCreate.size());
+
+          return succeeded(accountsToCreate);
+        }
+      ));
+  }
+
+  private boolean shouldChargeItemFee(LostItemPolicy policy) {
+    // Set cost fee is only supported now
+    return policy.getChargeAmountItem().getChargeType() == LostItemPolicyChargeAmountType.SET_COST
+      && isGreaterThanZero(policy.getChargeAmountItem().getAmount());
+  }
+
+  private boolean shouldChargeProcessingFee(LostItemPolicy policy) {
+    return policy.isChargeAmountItemPatron()
+      && isGreaterThanZero(policy.getLostItemProcessingFee());
+  }
+
+  private boolean shouldChargeAnyFee(LostItemPolicy policy) {
+    return shouldChargeItemFee(policy) || shouldChargeProcessingFee(policy);
+  }
+
+  private boolean isGreaterThanZero(BigDecimal numberToCompare) {
+    return numberToCompare != null && numberToCompare.compareTo(BigDecimal.ZERO) > 0;
+  }
+
+  private CompletableFuture<Result<FeeFineOwner>> getFeeFineOwner(Loan loan) {
+    final String servicePointId = loan.getItem().getLocation()
+      .getPrimaryServicePointId().toString();
+
+    return feeFineOwnerRepository.findOwnerForServicePoint(servicePointId);
+  }
+
+  private FeeFine getFeeFineOfType(Collection<FeeFine> feeFines, String type) {
+    return feeFines.stream()
+      .filter(feeFine -> feeFine.getFeeFineType().equals(type))
+      .findFirst()
+      .orElse(null);
+  }
+
+  private Result<Collection<FeeFine>> refuseWhenFeeFineOfTypeNotFound(
+    Result<Collection<FeeFine>> feeFinesResult, String type) {
+
+    return feeFinesResult.failWhen(
+      feeFines -> Result.succeeded(getFeeFineOfType(feeFines, type) == null),
+      feeFines -> singleValidationError("Expected automated fee of type " + type,
+        "feeFineType", type));
+  }
```

Comment:
I've split validation and FeeFine lookup from a collection because it simplifies code a bit. Otherwise I'll have to wrap the return value of  `getFeeFineOfType` into `Result`, which makes thinks a bit complicated.

But I'll try to think how these two methods can be combined.

Reply:
`getFeeFineOfType` is combined now with validation.